ui/plugin-system/src/runtime/DataQueriesProvider/DataQueriesProvider.tsx

```diff
@@ -31,47 +32,66 @@ export function useDataQueriesContext() {
   return ctx;
 }
 
+export function useDataQueries<T extends keyof QueryType>(queryType: T): UseDataQueryResults<QueryType[T]> {
+  const ctx = useDataQueriesContext();
+
+  // Filter the query results based on the specified query type
+  const filteredQueryResults = ctx.queryResults.filter(
+    (queryResult) => queryResult.definition.kind === queryType
+  ) as Array<QueryData<QueryType[T]>>;
+
+  // Filter the errors based on the specified query type
+  const filteredErrors = ctx.errors.filter((errors, index) => ctx.queryResults[index]?.definition.kind === queryType);
+
+  // Create a new context object with the filtered results and errors
+  const filteredCtx = {
+    queryResults: filteredQueryResults,
+    isFetching: filteredQueryResults.some((result) => result.isFetching),
+    isLoading: filteredQueryResults.some((result) => result.isLoading),
+    refetchAll: ctx.refetchAll,
+    errors: filteredErrors,
+  };
+
+  return filteredCtx;
+}
+
 export function DataQueriesProvider(props: DataQueriesProviderProps) {
   const { definitions, options, children } = props;
 
-  // For now we will map each query plugin definition to TimeSeriesQueryDefinition
-  // Later on when we add support for other query types,
-  // we will have to map each query maps to the correct QueryDefinition
-  const timeSeriesQueries = definitions.map(
-    (definition) =>
-      ({
+  const queryDefinitions = definitions.map((definition) => {
+    if (definition.type === undefined || definition.type === 'TimeSeriesQuery') {
```

Comment:
If these are explicitly `TimeSeriesQueries`, would it makes sense to adjust the variable name or is the idea that this conditional will be adjusted to include other types when they are added?

Reply:
> is the idea that this conditional will be adjusted to include other types when they are added? 

yep this!

Reply:
> could we look up the type via the kind of query?

@saminzadeh yep added getQueryType and removed `type` from the DataQueriesProvider definitions prop